src/Conf.ml

```diff
@@ -136,9 +136,37 @@ end = struct
 
   let store = ref []
 
+  let generated_doc ~names ~all ~allow_inline =
+    let name = List.hd_exn names in
+    let values =
+      let f acc (x, _) =
+        if String.is_empty acc then Format.sprintf "$(b,%s)" x
+        else Format.sprintf "%s,$(b,%s)" acc x
+      in
+      match all with
+      | [] -> user_error "config_file_doc needs at least one value" []
+      | [(x, _)] -> x
+      | _ -> Format.sprintf "{%s}" (List.fold all ~init:"" ~f)
+    in
+    let article =
+      match name.[0] with
+      | 'a' | 'e' | 'i' | 'o' | 'u' | 'y' -> "an"
+      | _ -> "a"
+    in
+    let config_file =
+      Format.sprintf
+        "Can be set in a config file with %s '$(b,%s) = %s' line." article
+        name values
+    in
+    let in_attributes =
+      if allow_inline then "" else " Cannot be set in attributes."
+    in
+    Format.sprintf "@,%s@,%s" config_file in_attributes
```

Comment:
I suspect that it would be easier for users if we had a separate section of the man page before the documentation of the options that stated that unless otherwise noted, any option `--option=value` can be set in the config file with an `option=value` line, or with the appropriate attribute. And then for the individual options, we ought to make sure to document all the possible values, e.g. by using `snd Cmdliner_base.enum` (as done for cmdliner's default `--help` option), and also to note the options that can _not_ be set in config files or attributes.